src/components/structures/RoomStatusBar.tsx

```diff
@@ -142,6 +166,30 @@ export default class RoomStatusBar extends React.PureComponent<IProps, IState> {
         dis.fire(Action.FocusSendMessageComposer);
     };
 
+    private onRefreshTimelineClick = async (): Promise<void> => {
+        this.setState({
+            isRefreshing: true,
+            refreshError: null,
+        });
+        try {
+            // Empty out the current timeline and re-request it
+            await this.props.room.refreshLiveTimeline();
+
+            this.setState({
+                timelineNeedsRefresh: false,
+            });
+        } catch (err) {
+            logger.error('Error while refresing the timeline:', err);
+            this.setState({
+                refreshError: err,
+            });
```

Comment:
Won't this lead to a weird state saying isRefreshing=true refreshError=error

Reply:
The `finally` will set `isRefreshing: false` which runs regardless of if the refresh succeeds or fails so we always end up in a settled state of not "loading".